capstone/cite/views.py

```diff
@@ -179,12 +179,19 @@ def case_pdf(request, case_id, pdf_name):
         applied before we return the case.
     """
     # check that we are at the canonical URL
-    case = get_object_or_404(CaseMetadata.objects.select_related('volume').prefetch_related('citations'), pk=case_id)
+    case = get_object_or_404(
+        CaseMetadata.objects.select_related('volume').prefetch_related('citations'),
+        pk=case_id
+    )
     pdf_url = case.get_pdf_url(with_host=False)
     if iri_to_uri(request.path) != pdf_url and not request.GET.get('redirect'):
-        return HttpResponseRedirect(pdf_url+"?redirect=1")
+        try:
+            return HttpResponseRedirect(pdf_url+"?redirect=1")
+        except TypeError:
+            # there is no pdf_url
+            raise Http404
```

Comment:
Probably doesn't matter much here but I think you'll get more predictable behavior explicitly checking for whether this variable is `None` and raising earlier, rather than inferring the problem from `TypeError`.

Reply:
Thanks. I wasn't sure what was most idiomatic/most directly applicable to the error in question. I think I'll check for `None`.